flocker/provision/_aws.py

```diff
@@ -60,14 +60,16 @@ def _wait_until_running(instance):
     with start_action(
         action_type=u"flocker:provision:aws:wait_until_running",
         instance_id=instance.id,
-    ):
-        while instance.state != 'running':
+    ) as context:
+        while instance.state != u'running' and instance.state != u'terminated':
             with start_action(
                 action_type=u"flocker:provision:aws:wait_until_running:sleep",
                 instance_state=instance.state,
             ):
                 sleep(1)
             instance.update()
+        context.add_success_fields(instance_state=instance.state)
+        return instance.state == u'running'
```

Comment:
Consider an exception for this case.  The function has failed to do what it is primarily meant to do (block further execution until the instance is running).  An exception is a good way to encourage the caller to handle that unexpected outcome (whereas return values can be ignored by a careless caller).